channeldb/invoice_test.go

```diff
@@ -119,12 +119,55 @@ func TestInvoiceWorkflow(t *testing.T) {
 		t.Fatalf("lookup should have failed, instead %v", err)
 	}
 
+	// Add an invoice with an external preimage
+	// Create a fake invoice with an external preimage which we'll use several times in the tests
+	// below.
+	fakeExternalPreimageInvoice := &Invoice{
+		// Use single second precision to avoid false positive test
+		// failures due to the monotonic time component.
+		CreationDate: time.Unix(time.Now().Unix(), 0),
+	}
+	fakeExternalPreimageInvoice.Memo = []byte("memo")
+	fakeExternalPreimageInvoice.Receipt = []byte("receipt")
+	fakeExternalPreimageInvoice.PaymentRequest = []byte("")
+	fakeExternalPreimageInvoice.Terms.ExternalPreimage = true
+	externalPreimagePaymentHash := sha256.Sum256([]byte("fake preimage"))
+	fakeExternalPreimageInvoice.Terms.PaymentHash = externalPreimagePaymentHash
+	fakeExternalPreimageInvoice.Terms.Value = lnwire.NewMSatFromSatoshis(10000)
+
+	// Add the invoice to the database, this should succeed as there aren't
+	// any existing invoices within the database with the same payment
+	// hash.
+	if err := db.AddInvoice(fakeExternalPreimageInvoice); err != nil {
+		t.Fatalf("unable to add invoice: %v", err)
+	}
+
+	// Attempt to retrieve the invoice which was just added to the
+	// database. It should be found, and the invoice returned should be
+	// identical to the one created above.
+	dbExternalPreimageInvoice, err := db.LookupInvoice(externalPreimagePaymentHash)
+	if err != nil {
+		t.Fatalf("unable to find invoice: %v", err)
+	}
+	if !reflect.DeepEqual(fakeExternalPreimageInvoice, dbExternalPreimageInvoice) {
+		t.Fatalf("invoice fetched from db doesn't match original %v vs %v",
+			spew.Sdump(fakeExternalPreimageInvoice), spew.Sdump(dbExternalPreimageInvoice))
```

Comment:
Does this log show up correctly if its wrong? (id imagine this is a huge log)

Reply:
Yes, but only for failed tests cases.